pkg/client/observatorium/api_test.go

```diff
@@ -2,9 +2,62 @@ package observatorium
 
 import (
 	"testing"
+
+	. "github.com/onsi/gomega"
 )
 
+func Test_GetKafkaState(t *testing.T) {
+	RegisterTestingT(t)
+	type fields struct {
+		client *Client
+	}
+
+	type args struct {
+		name              string
+		resourceNamespace string
+	}
+
+	obsClientMock, err := NewClientMock(&Configuration{})
+	Expect(err).To(BeNil())
+
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+		want    KafkaState
+	}{
+		{
+			name: "should return no error and Ready KafkaState",
+			fields: fields{
+				client: obsClientMock,
+			},
+			args: args{
+				name:              "test",
+				resourceNamespace: "test",
+			},
+			wantErr: false,
+			want: KafkaState{
+				State: ClusterStateReady,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			obs := &ServiceObservatorium{
+				client: tt.fields.client,
+			}
+			state, err := obs.GetKafkaState(tt.args.name, tt.args.resourceNamespace)
+			Expect(err != nil).To(Equal(tt.wantErr))
+			Expect(tt.want).To(Equal(state))
```

Comment:
Just a small comment:
```diff
- Expect(tt.want).To(Equal(state))
+ Expect(state).To(Equal(tt.want))
```

Otherwise lgtm, but I'd wait for someone else to have a look too :)